modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/MiniCart.js

```diff
@@ -108,61 +132,148 @@ function MiniCart({
 				updateCartState,
 			}}
 		>
-			{!!CartView && (
-				<div className={classnames('mini-cart', isOpen && 'is-open')}>
-					<div
-						className={'mini-cart-overlay'}
-						onClick={() => setIsOpen(false)}
-					/>
+			{!!CartViews[CART] && (
+				<div
+					className={classnames(
+						'mini-cart',
+						(!toggleable || isOpen) && 'is-open'
+					)}
+				>
+					{toggleable && (
+						<>
+							<div
+								className={'mini-cart-overlay'}
+								onClick={() => setIsOpen(false)}
+							/>
 
-					<Opener openCart={openCart} />
+							<CartViews.Opener openCart={openCart} />
+						</>
+					)}
 
-					<CartView cartItemsListView={cartItemsListView} />
+					<CartViews.Cart />
 				</div>
 			)}
 		</MiniCartContext.Provider>
 	);
 }
 
 MiniCart.defaultProps = {
-	cartItemsListView: {
-		component: CartItemsList,
-	},
-	cartView: {
-		component: Wrapper,
-	},
+	cartViews: DEFAULT_VIEWS,
 	displayDiscountLevels: false,
+	displayTotalItemsQuantity: false,
+	itemsQuantity: 0,
+	labels: DEFAULT_LABELS,
+	onAddToCart: () => {},
+	summaryDataMapper,
+	toggleable: true,
 };
 
 MiniCart.propTypes = {
 	cartActionURLs: PropTypes.shape({
 		checkoutURL: PropTypes.string,
 		orderDetailURL: PropTypes.string,
 	}).isRequired,
-	cartItemsListView: PropTypes.shape({
-		component: PropTypes.func,
-		contentRendererModuleUrl: PropTypes.string,
-	}),
-	cartView: PropTypes.shape({
-		component: PropTypes.func,
-		contentRendererModuleUrl: PropTypes.string,
+	cartViews: PropTypes.shape({
+		[CART]: PropTypes.oneOfType([
+			PropTypes.shape({
+				component: PropTypes.func,
+			}),
+			PropTypes.shape({
+				contentRendererModuleUrl: PropTypes.string,
+			}),
+		]),
+		[HEADER]: PropTypes.oneOfType([
+			PropTypes.shape({
+				component: PropTypes.func,
+			}),
+			PropTypes.shape({
+				contentRendererModuleUrl: PropTypes.string,
+			}),
+		]),
+		[ITEM]: PropTypes.oneOfType([
+			PropTypes.shape({
+				component: PropTypes.func,
+			}),
+			PropTypes.shape({
+				contentRendererModuleUrl: PropTypes.string,
+			}),
+		]),
+		[ITEMS_LIST]: PropTypes.oneOfType([
+			PropTypes.shape({
+				component: PropTypes.func,
+			}),
+			PropTypes.shape({
+				contentRendererModuleUrl: PropTypes.string,
+			}),
+		]),
+		[ITEMS_LIST_ACTIONS]: PropTypes.oneOfType([
+			PropTypes.shape({
+				component: PropTypes.func,
+			}),
+			PropTypes.shape({
+				contentRendererModuleUrl: PropTypes.string,
+			}),
+		]),
+		[OPENER]: PropTypes.oneOfType([
+			PropTypes.shape({
+				component: PropTypes.func,
+			}),
+			PropTypes.shape({
+				contentRendererModuleUrl: PropTypes.string,
+			}),
+		]),
+		[ORDER_BUTTON]: PropTypes.oneOfType([
+			PropTypes.shape({
+				component: PropTypes.func,
+			}),
+			PropTypes.shape({
+				contentRendererModuleUrl: PropTypes.string,
+			}),
+		]),
+		[SUMMARY]: PropTypes.oneOfType([
+			PropTypes.shape({
+				component: PropTypes.func,
+			}),
+			PropTypes.shape({
+				contentRendererModuleUrl: PropTypes.string,
+			}),
+		]),
```

Comment:
Sure will be nice to replace all this with TypeScript one day...

Reply:
For sure it will :D